Wendy/assignment1Answers.html

```diff
@@ -0,0 +1,219 @@
+<!doctype html>
+<html class="no-js" lang="">
+    <head>
+        <meta charset="utf-8">
+        <meta http-equiv="x-ua-compatible" content="ie=edge">
+        <meta name="viewport" content="width=device-width, initial-scale=1">
+        <title>JavaScript Immersion Assignment 1 || Wendy Goodwin</title>
+        <link href="css/bootstrap.min.css" rel="stylesheet">
+	    <link href="css/main.css" rel="stylesheet">    
+	    <link href="css/mystyles.css" rel="stylesheet">
+	</head>
+
+	<body>
+
+		<div id="wrap">
+			<div class="title">
+				<h1> JavaScript Immersion Assignment I </h1>
+				<h3> by Wendy Goodwin </h3>	
+				<h4> 
+
+					<a href="https//github/portlandcodeschool/jse=fall15-1">	View assignment specifications</a>
+
+
+				</h4>
+			</div>
+			<div class="row" id="part1">
+				<h3> Part I </h3>
+				<div class="col-md-4">
+					<h3>a) "1" == 1  true.</h3>
+					<p>The == equality operator says "kind of the same" and they are.  Javascript reads the equality operator from right to left and changes the operators to the same type if needed.  
+
+					<h3>b) "1" === 1 false.</h3>
+					<p>The "===" is "exactly equal to"  No room for kind of.</p>
+
+					<h3>c) x == 'x'  false.</h3>
+					<p>  If x is defined as a string, the 'x' is saying string 'x' and if your string is for example var x = 'snowman', 'snowman' != 'x'.  If x is a number, it is still false because a number is not a string.</p>
+
+
+					<h3>d) x == (x+'') false.</h3>
+					<p>This is false because the '' empty quotes make things falsy!</p>
+
+					<h3>e) '' == ' ' false.</h3>  Nothing is not equal to a space.  A space is something.</h3>
+
+					<h3>f) x = true  true.</h3>
+					<p>  Assigning the boolean value of true to x.</p>
+				</div>
+
+				<div class="col-md-4">
+					<h3>g) var x; x == 'undefined'  false.</h3> 
+					<p>It seems like the intent was to initialize a variable but leave it as undefined.  However, variables are already undefined until they are initialzed.  Moreover, the =='undefined' is shown as a string not a value.  Undefined would be a value not a string.</p>  
+
+					<h3>h) '9'<'10' false.</h3>
+					<p>  '' makes these strings not numbers and so they inherently do not have values.</p>
```

Comment:
Ah, but indeed they do have value! The < comparator is checking '9' vs '10' one character at a time. In this situation '9' is greater than '1' so javascript stops comparison and returns false. Similarly, '950' < '9256' returns false as well, because '9' == '9', which moves to the next character set, returning '5' > '2'. As soon as javascript finds one character greater than another, it discontinues comparing the sets, and returns with a boolean. This works with alpha strings as well. For example, 'b' > 'a’!